test/models/_model_test_case.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import absolute_import
```

Comment:
Love this!!!

Reply:
It might be good to see if we could make this the superclass for most of our tests in `test/examples`. We could add a training loop in these tests, and maybe we can add a flag that optionally turns off the training loop tests if we just want a fast run of the test suite.